cabal-testsuite/main/cabal-tests.hs

```diff
@@ -238,7 +238,21 @@ main = do
             -- Simple runner
             (real_path, real_args) <- runTest (runnerCommand senv) path
             hPutStrLn stderr $ showCommandForUser real_path real_args
-            callProcess real_path real_args
+            -- If the test was reported flaky, the `runghc` call will exit
+            -- with exit code 1, and report `TestCodeFlaky` on the stderr output
+            --
+            -- This seems to be the only way to catch this case.
+            --
+            -- Sadly it means that stdout and stderr are not interleaved
+            -- directly anymore.
+            (e, out, err) <- readProcessWithExitCode real_path real_args ""
+            putStrLn "# STDOUT:"
+            putStrLn out
+            putStrLn "# STDERR:"
+            putStrLn err
+            if "TestCodeFlaky" `isInfixOf` err
+              then pure ()
+              else throwIO e
```

Comment:
This grepping through stdout feels a little crude although I don't know a better way. Also, usually, when you switch from showing direct output to showing captured one, there are some side effects. E.g. colored output would be unavailable, etc. Of course, Cabal testsuite doesn't have colored output in particular but there may be spooky actions at the distance...

Reply:
I thought maybe there was some other way to pipe this output of the process but I didn't seem to find one.

In any case this is only for when you invoke the tests with one test. I don't know how much that is used, I personally run the validate script always

Reply:
> this is only for when you invoke the tests with one test

Oh, that's much less of a concern then. Thank you!

Reply:
Let's leave it as unresolved for now maybe to see if anyone has other thoughts...